### PR TITLE
added to combineReducers and fixed products rendering bug

### PR DESCRIPTION
**Pull request by saba-mo**
### Assignee Tasks

* [ ] added all current reducers to combineReducer
* [ ] fixed all and single products rendering bug 

### Guidelines

Please add a description of this Pull Request's motivation, scope, outstanding issues or potential alternatives, reasoning behind the current solution, and any other relevant information for posterity.

---

_Your PR Notes Here_

### CODE REVIEW

File: client/components/all-products.js [hunk 1]
```diff
@@ -3,14 +3,15 @@ import {Link} from 'react-router-dom'
...
+  async componentDidMount() {
+    await this.props.getProducts()
   }
 
-  //is the link correct? Check
-
   render() {
+    if (this.props.products === undefined) {
+      return <div>Loading...</div>
+    }
```
Comment (ahillary) [inline client/components/all-products.js]: can delete, now that test is done

File: client/components/all-products.js [hunk 2]
```diff
@@ -3,14 +3,15 @@ import {Link} from 'react-router-dom'
 import {connect} from 'react-redux'
 import {fetchProducts} from '../store/products'
 
-export class AllProducts extends React.Component {
-  componentDidMount() {
-    this.props.getProducts()
+class AllProducts extends React.Component {
+  async componentDidMount() {
+    await this.props.getProducts()
```
Comment (ahillary) [inline client/components/all-products.js]: check

File: client/components/single-product.js [hunk 3]
```diff
@@ -2,15 +2,14 @@ import React from 'react'
...
+class SingleProduct extends React.Component {
   // constructor() {
   //   super()
 
   //   // this.handleAddToCart = this.handleAddToCart.bind(this)
   // }
 
   async componentDidMount() {
-    console.log('mount: ', this.props)
     await this.props.loadSingleProduct(this.props.match.params.productId)
```
Comment (ahillary) [inline client/components/single-product.js]: check async wait

**Review by ahillary — CHANGES_REQUESTED** [review]
see comments. Can delete some lines, and check on others

**Review by abstauff — APPROVED** [review]

**Review by ahillary — APPROVED** [review]

**Pull request merged**